scripts/linters/js_ts_linter.py

```diff
@@ -284,6 +291,139 @@ def _check_http_requests(self):
+                                previous_line_has_ts_ignore = False
+                                continue
+
+                        failed = True
+                        previous_line_has_ts_ignore = False
+                        summary_message = (
+                            '%s --> @ts-ignore found at line %s. '
+                            'Please add this exception in %s.' % (
+                                file_path, line_number,
+                                TS_IGNORE_EXCEPTIONS_FILEPATH))
+                        python_utils.PRINT(summary_message)
+                        summary_messages.append(summary_message)
+                        python_utils.PRINT('')
+
+                    previous_line_has_ts_ignore = bool(
+                        re.findall(ts_ignore_pattern, line))
+
+                    if (
+                            previous_line_has_ts_ignore and
+                            not previous_line_has_comment_with_ts_error):
+                        failed = True
+                        summary_message = (
+                            '%s --> Please add a comment above the @ts-ignore '
+                            'explaining the @ts-ignore at line %s. The format '
+                            'of comment should be -> This throws "...". This '
+                            'needs to be suppressed because ...' % (
+                                file_path, line_number + 1))
+                        python_utils.PRINT(summary_message)
+                        summary_messages.append(summary_message)
+                        python_utils.PRINT('')
+
+                    previous_line_has_comment = bool(
+                        re.findall(comment_pattern, line))
+
+                    previous_line_has_comment_with_ts_error = (
+                        bool(
+                            re.findall(
+                                comment_with_ts_error_pattern, line))
+                        or (
+                            previous_line_has_comment_with_ts_error and
+                            previous_line_has_comment))
+
+            if failed:
+                summary_message = (
+                    '%s TS ignore check failed' % (
+                        linter_utils.FAILED_MESSAGE_PREFIX))
+            else:
+                summary_message = (
+                    '%s TS ignore check passed' % (
+                        linter_utils.SUCCESS_MESSAGE_PREFIX))
+
+            python_utils.PRINT(summary_message)
+            summary_messages.append(summary_message)
+            python_utils.PRINT('')
+
+        return summary_messages
+
+    def _check_ts_expect_error(self):
+        """Checks if ts expect error is used in non spec file."""
+        summary_messages = []
+        if self.verbose_mode_enabled:
+            python_utils.PRINT('Starting ts expect error check')
+            python_utils.PRINT('----------------------------------------')
+
+        ts_expect_error_pattern = r'@ts-expect-error'
```

Comment:
Require specification of the error message, similar to ts-ignore.

Reply:
done